scripts/tabs.js

```diff
@@ -26,45 +29,65 @@ function capitalize (languageName) {
   return languageName.charAt(0).toUpperCase() + languageName.slice(1);
 }
 
-function appendLanguageBlock (tabEl, preEl, languageName, id, active) {
-  const navTag = tabEl.find('.nav');
-  navTag.append(`<li role="presentation" ${active ? 'class="active"' : ''} role="tab">
-    <a href='#${id}' data-toggle='tab'>${capitalize(languageName)}</a>
-  </li>`);
-  const tabPanel = tabEl.find('.tab-content');
-  let html = preEl[0].outerHTML;
-  html = stripLanguageComment(html);
-  tabPanel.append(`<div role="tabpanel" class="tab-pane ${active ? 'active' : ''}" id="${id}">${html.trim()}</div>`);
-}
-
 export function fencedCodeTabify (html) {
   const jqHTML = $(html);
-  let tabTagHTML = `<div>
-    <ul class="nav nav-tabs" role="tablist">
-    </ul>
-    <div class="tab-content">
-    </div>
-  </div>`;
 
   let tabPaneIndex = 0;
 
+  let languageBlocks = [];
+
   jqHTML.find("pre > code").each((index, codeTag) => {
     const preTag = $(codeTag).parent();
     const siblings = preTag.nextAll();
-    const tabEl = $(tabTagHTML);
     let language = capitalize($(codeTag).attr('class'));
-    let siblingIndex = 0;
-    appendLanguageBlock(tabEl, preTag, language, `${tabPaneIndex}_${siblingIndex++}`, true);
+
+    let siblingCount = 1;
+    languageBlocks.push({
+      language: language.toLowerCase(),
```

Comment:
I'd rather move this to a separate function to avoid unnecessary code duplication:

```js
const pushLanguageBlock = (lang, el, tabPaneIndex) => {
   languageBlocks.push({
         language: lang.toLowerCase(),
         capitalizedLanguage: capitalize(lang),
         html: stripLanguageComment($(el)[0].outerHTML),
         tabIndex,
       });
}
...
```